doc/architecture/decisions/0019-BLAST-database-files.md

```diff
@@ -0,0 +1,41 @@
+# 19. BLAST database files
+
+Date: 2022-09-11
+
+## Status
+
+Draft
+
+## Context
+
+This document summarises the decisions made when choosing which database options to support for the initial implementation of BLAST on the new Ensembl website.  
+
+## Decision
+
+### Database choices
+BLAST databases are generated from FASTA files copied from the Ensembl and ensemblgenomes FTP sites
+
+#### Genomic sequence
+Unmmasked and softmasked masked, excluding hardmasked. Softmasked sequence has all the sequence, and so the coords are consistent, it’s just that parts are lowercase and will be ignored by BLAST when it’s looking for regions to seed the alignments. An alignment seed can be found near a repeat and the alignment can then be extended into the repeat though. This is based on the idea that you’re probably not looking for things that are wholly contained in repeats. Plus it makes BLAST much faster as it looks at less of the genome.
+Hardmasked sequence has all the repeat regions removed, so it’s essentially like a very extreme version of softmasking that removes anything that looks like a repeat and also changes the coordinates as a result. As a result there will be no alignments to anything classed as a repeat, even through extension of the initial alignment. There are very few scenarios where this would be a good thing and potentially it’s very confusing and misleading.
+
+#### Transcripts
+cDNAs containing sequences of all transcripts, from both protein coding and non coding genes. Proteins for all protein coding transcripts.
+We chose not to support ab initio transcripts, an option available on the current website since predicting transcripts using these methods, for example by SNAP, is very much an outdated technology.
```

Comment:
> an option available on the current website

This will become meaningless to the future us when the new website becomes the current website.

Reply:
I don't think it'll be meaningless, and we will have an archived code base even after the sites are finally retired.